samtranslator/model/sam_resources.py

```diff
@@ -2407,3 +2419,195 @@ def _set_resolver_code_settings(resolver_code_settings: aws_serverless_graphqlap
+
+        none_datasource_logical_id = f"{self.logical_id}NoneDataSource"
+        none_datasource = self._check_and_construct_none_datasource(functions, api_id, none_datasource_logical_id)
+
+        if none_datasource:
+            resources.append(none_datasource)
+
+        for relative_id, function in functions.items():
+            func_config = FunctionConfiguration(
+                logical_id=self.logical_id + relative_id,
+                depends_on=self.depends_on,
+                attributes=self.resource_attributes,
+            )
+
+            # "Id" refers to the "LogicalId" attribute for a "AppSync::FunctionConfiguration" resource.
+            # "Id" is a mutually exclusive property to every other property. If this function has it
+            # defined, then we make sure it's the only property, and continue to next function.
+            if function.Id:
+                keys = remove_none_values(function.dict()).keys()  # remove undefined properties, then get keys
+                if len(keys) != 1:
+                    raise InvalidResourceException(
+                        relative_id, "'Id' cannot be defined with other properties in Function."
+                    )
+                continue
+
+            func_config.ApiId = api_id
+            func_config.Name = function.Name or relative_id
+            func_config.Code, func_config.CodeS3Location = self._parse_code_properties(
+                function, resolver_code_settings, func_config.Name, relative_id
+            )
+            func_config.DataSourceName = self._parse_datasource_name(
+                relative_id, function, datasources, none_datasource_logical_id
+            )
+            func_config.MaxBatchSize = passthrough_value(function.MaxBatchSize)
+            func_config.Description = passthrough_value(function.Description)
+            func_config.Runtime = self._parse_runtime(function, resolver_code_settings, relative_id)
+
+            if function.Sync:
+                func_config.SyncConfig = cast(SyncConfigType, function.Sync.dict())
+
+            resources.append(func_config)
+
+        return resources
+
+    def _check_and_construct_none_datasource(
+        self,
+        functions: Dict[str, aws_serverless_graphqlapi.Function],
+        api_id: Intrinsicable[str],
+        none_datasource_logical_id: str,
+    ) -> Optional[DataSource]:
+        """
+        Checks if a DataSource with type "NONE" needs to be created.
+
+        Within a Serverless::GraphQLApi Function resource, customers can create a
+        DataSource of Type "NONE" for quick use in Functions. To do so, a customer
+        can input "none" case insensitive in the "DataSource" property. Only one
+        DataSource will be created for each GraphQLApi, and all GraphQLApi functions
+        with this none input will reference it.
+        """
+        for function in functions.values():
+            if not function.DataSource or function.DataSource.lower() != "none":
```

Comment:
This would be clearer as its own function (i.e. take `function` as input, return boolean whether it's none data source) since it contains some business logic ("case-insensitive `none`") and is repeated in other parts of the PR.

Reply:
```
def _is_none_datasource_input(datasource: str):
    return True if datasource and datasource.lower() == "none" else False
```

Something like this?

Reply:
Yeah, maybe take the `function` as input so you have one less thing to check, and it kinda makes sense in its own function.

Reply:
Not sure what you mean here. What one less thing do I check by passing in the whole function object?

Reply:
Ah, ignore, read the condition wrong. Your example makes sense (except get rid of the unnecessary ternary).